### DISCUSSION TITLE
[General] `Validation revert: Sender is not an account`

### DISCUSSION THREAD

**Discussion by PatrickAlphaC** [zkSync-Community-Hub/zksync-developers#570] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

I have a minimal custom account abstraction based smart contract on zkSync here: [0x1Ec2090975a6a497935891c25E7535893D9FEF7e](https://explorer.zksync.io/address/0x1Ec2090975a6a497935891c25E7535893D9FEF7e). Let's call it `zkAccount` moving forward. 

I'm trying to have the following flow occur:
1. I build a transaction where `zkAccount` approves an address to spend `1e6` USDC
2. I send the transaction as a type `113`, and my `zkAccount` smart contract pays for it, using zkSync's native AA

I have a codebase in typescript that is supposed to be able to send such a transaction:

```javascript
    aaTx = {
        ...aaTx,
        from: ZK_MINIMAL_ADDRESS,
        gasLimit: gasLimit,
        gasPrice: gasPrice,
        chainId: (await provider.getNetwork()).chainId,
        nonce: await provider.getTransactionCount(ZK_MINIMAL_ADDRESS),
        type: 113,
        customData: {
            gasPerPubdata: utils.DEFAULT_GAS_PER_PUBDATA_LIMIT,
        } as types.Eip712Meta,
        value: ethers.getBigInt(0),
    }
    const signedTxHash = EIP712Signer.getSignedDigest(aaTx)

    const signature = ethers.concat([
        ethers.Signature.from(wallet.signingKey.sign(signedTxHash)).serialized,
    ])

    aaTx.customData = {
        ...aaTx.customData,
        customSignature: signature,
    }

    const sentTx = await provider.broadcastTransaction(
        types.Transaction.from(aaTx).serialized,
    )
```

However, every time I run it, I run into the following:

```
Error: could not coalesce error (error={ "code": 3, "data": "0x", "message": "failed to validate the transaction. reason: Validation revert: Sender is not an account" }, payload={ "id": 13, "jsonrpc": "2.0", "method": "eth_sendRawTransaction", "params": [ "SOME_PARAMS_HERE" ] }, code=UNKNOWN_ERROR, version=6.13.0)
    at makeError (/Path/minimal-account-abstraction/node_modules/ethers/src.ts/utils/errors.ts:694:21)
    at Provider.getRpcError (/Path/minimal-account-abstraction/node_modules/ethers/src.ts/providers/provider-jsonrpc.ts:1059:25)
    at /Path/minimal-account-abstraction/node_modules/ethers/src.ts/providers/provider-jsonrpc.ts:563:45
    at processTicksAndRejections (node:internal/process/task_queues:95:5) {
  code: 'UNKNOWN_ERROR',
  error: {
    code: 3,
    message: 'failed to validate the transaction. reason: Validation revert: Sender is not an account',
    data: '0x'
  },
  payload: {
    method: 'eth_sendRawTransaction',
    params: [
      '<SOME_PARAMS_HERE>'
    ],
    id: 13,
    jsonrpc: '2.0'
  },
  shortMessage: 'could not coalesce error'
}
```

I have seen the following warning:

> To deploy the multisig smart contract, it is necessary to interact with the `DEPLOYER_SYSTEM_CONTRACT` and call the `create2Account` function. If the code doesn't do this, you may see errors like- `Validation revert: Sender is not an account.`

I've seen this on other tutorials, however this does not apply to me, as I am not deploying my smart contract, or any contract.

Why am I getting this error? 

### Repo Link (Optional)

https://github.com/PatrickAlphaC/minimal-account-abstraction/tree/cyfrin-mock-era-contracts

**Comment by PatrickAlphaC** [comment]
Looks like this is what's firing. I'm not sure why though. Wouldn't this make native account abstraction worthless?

https://github.com/dto-simba/zksync-era/blob/da6bcf2e1c9122dadfb14ec780f6c1051f3bd437/core/lib/vm/src/errors/tx_revert_reason.rs#L173

**Comment by PatrickAlphaC** [comment]
Ah... I think I got it.

When you deploy, that's the only time you can signify a contract is an AA contract... I think I got it.

**Comment by uF4No** [comment, marked as answer]
>When you deploy, that's the only time you can signify a contract is an AA contract.

Correct.

Was the account you're using as a `from` deployed using `createAccount` or `create2Account`?

**Reply by PatrickAlphaC** [reply]
It was deployed using `create` I didn't realize that was how AA accounts had to be deployed. Makes sense now.

**Reply by 4NNNN** [reply]
If it's a contract it works but if it's create2Account deployed proxy it doesn't work. Refer [639](https://github.com/zkSync-Community-Hub/zksync-developers/discussions/639)
